lib/paperclip/io_adapters/data_uri_adapter.rb

```diff
@@ -10,10 +12,74 @@ def initialize(target_uri)
     private
 
     def extract_target(uri)
-      data_uri_parts = uri.match(REGEXP) || []
-      StringIO.new(Base64.decode64(data_uri_parts[2] || ''))
+      @data_uri_parts = (uri.match(REGEXP) || []).to_a
+
+      @payload = get_payload_from_match
```

Comment:
For an example of what that might look like, you could check out [this parser](https://github.com/timsly/carrierwave-data-uri/blob/master/lib/carrierwave_data_uri/parser.rb) @timsly wrote for his carrierwave-data-uri plugin.

I don't know if there's a rivalry between Paperclip and Carrierwave, but it seems like there might be an opportunity to combine efforts on Data URI handling here.